### PR TITLE
Add WZRY protocol dissector and ndpiReader error tips when pcap file open failed.

### PR DESCRIPTION
**Pull request by Gowa2017**
When open a device, we do not know why we failed.

### CODE REVIEW

File: src/lib/protocols/wzry.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#include "ndpi_protocol_ids.h"
+
+#define NDPI_CURRENT_PROTO NDPI_PROTOCOL_WZRY
+
+#include "ndpi_api.h"
+
+static void
+ndpi_int_wzry_add_connection(struct ndpi_detection_module_struct *ndpi_struct,
+                             struct ndpi_flow_struct             *flow /* , */
+                             /* ndpi_protocol_type_t protocol_type */) {
```
Comment (IvanNardi) [inline src/lib/protocols/wzry.c:10]: I know that you have probably copied this prototype from some other file, but you can completely remove the third parameter

File: src/lib/protocols/wzry.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+void ndpi_search_wzry(struct ndpi_detection_module_struct *ndpi_struct,
+                      struct ndpi_flow_struct             *flow) {
+    struct ndpi_packet_struct *packet = &ndpi_struct->packet;
+
+    NDPI_LOG_DBG(ndpi_struct, "search WZRY\n");
+
+    fprintf(stderr,"%d,%d\n",flow->l4_proto, IPPROTO_TCP);
+    fprintf(stderr,"%s -> %d",flow->saddr, flow->saddr);
```
Comment (IvanNardi) [inline src/lib/protocols/wzry.c:22]: All these fprintf calls should be removed

File: src/lib/protocols/wzry.c [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+
+    if (flow->l4_proto == IPPROTO_TCP && packet->l3_packet_len > 4 &&
+        get_u_int8_t(packet->payload, 0) == 0x33 &&
+        get_u_int8_t(packet->payload, 1) == 0x66 &&
+        get_u_int8_t(packet->payload, 2) == 0x00 &&
+        get_u_int8_t(packet->payload, 3) == 0x09) {
+        NDPI_LOG_INFO(ndpi_struct, "found WZRY TCP\n");
+        fprintf(stderr,"found WZRY\n");
```
Comment (IvanNardi) [inline src/lib/protocols/wzry.c:31]: As before

File: src/lib/protocols/wzry.c [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+        ndpi_int_wzry_add_connection(ndpi_struct, flow);
+    } else if (flow->l4_proto == IPPROTO_UDP && packet->l3_packet_len > 4 &&
+               ntohl(get_u_int8_t(packet->payload, 0)) == 0x01 &&
+               ntohl(get_u_int8_t(packet->payload, 1)) == 0x02 &&
+               ntohl(get_u_int8_t(packet->payload, 2)) == 0x00 &&
+               ntohl(get_u_int8_t(packet->payload, 3)) == 0x00) {
+        NDPI_LOG_INFO(ndpi_struct, "found WZRY TCP\n");
+        fprintf(stderr,"found WZRY\n");
```
Comment (IvanNardi) [inline src/lib/protocols/wzry.c:40]: Same

File: src/lib/protocols/wzry.c [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    }
+}
+
+void init_wzry_dissector(struct ndpi_detection_module_struct *ndpi_struct,
+                         u_int32_t                           *id,
+                         NDPI_PROTOCOL_BITMASK *detection_bitmask) {
+    ndpi_set_bitmask_protocol_detection(
+        "WZRY", ndpi_struct, detection_bitmask, *id, NDPI_PROTOCOL_WZRY,
+        ndpi_search_wzry,
+        NDPI_SELECTION_BITMASK_PROTOCOL_V4_V6_TCP_OR_UDP_WITH_PAYLOAD,
```
Comment (IvanNardi) [inline src/lib/protocols/wzry.c:53]: I am pretty sure that `NDPI_SELECTION_BITMASK_PROTOCOL_V4_V6_TCP_OR_UDP_WITH_PAYLOAD_WITHOUT_RETRANSMISSION` is better

**Review by IvanNardi — CHANGES_REQUESTED** [review]
Could you add some traces as example with both TCP and UDP flows, please?

**Comment by Gowa2017** [comment]
I'sorry , I only want to push the error tip.

But  when I test WZRY protocol, I pushed tje test code, so I will close the pr.

**Pull request closed without merging**